pkg/api/norman/customization/aks/handler.go

```diff
@@ -154,6 +164,24 @@ func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credI
 	cap.ClientID = string(cc.Data["azurecredentialConfig-clientId"])
 	cap.ClientSecret = string(cc.Data["azurecredentialConfig-clientSecret"])
 
+	if cap.TenantID == "" {
+		cap.TenantID, err = aks.GetCachedTenantID(cap.SubscriptionID, cc)
+		if err != nil {
+			return httperror.ServerError.Status, err
+		}
+		cc.Annotations[tenantIDAnnotation] = cap.TenantID
+		cc.Annotations[tenantIDTimestampAnnotation] = time.Now().UTC().Format(time.RFC3339)
```

Comment:
Shouldn't this only be refreshed if FindTenantID was used in GetCachedTenantID why refresh the annotations if it was retrieved from the annotations?

Reply:
Fixed in aks-operator